isobus/src/can_extended_transport_protocol.cpp

```diff
@@ -799,9 +799,18 @@ namespace isobus
 	bool ExtendedTransportProtocolManager::send_clear_to_send(std::shared_ptr<ExtendedTransportProtocolSession> &session) const
 	{
 		std::uint32_t nextPacketNumber = session->get_last_packet_number() + 1;
+		std::uint32_t bytesProcessed = (PROTOCOL_BYTES_PER_FRAME * session->get_last_packet_number());
+		std::uint8_t numberOfPacketsToClear = session->get_cts_number_of_packet_limit();
+
+		if (((PROTOCOL_BYTES_PER_FRAME * numberOfPacketsToClear) + bytesProcessed) > (session->get_message_length()))
+		{
+			std::uint32_t bytesRemaining = session->get_message_length() - bytesProcessed;
+			numberOfPacketsToClear = (bytesRemaining / 7) + static_cast<std::uint8_t>(0 != (bytesRemaining % 7));
+		}
+
```

Comment:
```suggestion
		std::uint8_t packetLimit = session->get_cts_number_of_packet_limit();
		if (packetLimit > session->get_number_of_remaining_packets())
		{
			packetLimit = static_cast<std::uint8_t>(session->get_number_of_remaining_packets());
		}
```

This was my fix haha, might be a bit friendlier to look at?

Reply:
Not sure if it will actually fix it though, haven't tested it. But it should do the same

Reply:
I'll test it real quick - it does look nicer, I just didn't know that get_number_of_remaining_packets existed haha

Reply:
Yep, that works! Suggestion shamelessly stolen